packages/@aws-cdk/aws-s3/README.md

```diff
@@ -92,13 +92,36 @@ A bucket policy will be automatically created for the bucket upon the first call
 
 ```ts
 const bucket = new Bucket(this, 'MyBucket');
-bucket.addToResourcePolicy(new iam.PolicyStatement({
+const result = bucket.addToResourcePolicy(new iam.PolicyStatement({
```

Comment:
Note for the future -

I'm not a big fan of this behaviour. Ideally, this API should fail if the 'add' does not occur. It's feels counter intuitive, poor ergonomics, that an action 'add' doesn't actually 'add' in some cases.

We should have added a second API `tryAddToResourcePolicy()` that does not fail in such cases.

Reply:
Good point. I can create this new API.

Reply:
Wondering if this API has been created and whether this can fail more loudly! The silent failure of `addToResourcePolicy` is very confusing (per https://github.com/aws/aws-cdk/issues/6548) and it's not clear under what circumstances a person should know to check for success.